x/src/test.rs

```diff
@@ -26,7 +32,19 @@ pub struct Args {
 pub fn run(mut args: Args, config: Config) -> Result<()> {
     args.args.extend(args.testname.clone());
 
-    let cmd = CargoCommand::Test(&args.args);
+    let mut direct_args = Vec::new();
+    if args.no_run {
+        direct_args.push(OsString::from("--no-run"));
+    };
+    if args.no_fail_fast {
+        direct_args.push(OsString::from("--no-fail-fast"));
+    };
```

Comment:
Might want to make a note that this will only affect the individual runs of cargo that x does, and not the whole x command